sdk/state/ingest_test.go

```diff
@@ -191,9 +200,88 @@ func TestChannel_IngestTx_oldDeclTx(t *testing.T) {
 
 	// Pretend that responder broadcasts the old declTx, and
 	// initiator ingests it.
-	err = initiatorChannel.IngestTx(oldDeclTx)
+	err = initiatorChannel.IngestTx(oldDeclTx, dummyXDR)
 	require.NoError(t, err)
 	closeState, err := initiatorChannel.CloseState()
 	require.NoError(t, err)
 	require.Equal(t, CloseStateClosingWithOutdatedState, closeState)
 }
+
+func TestChannel_IngestTx_updateBalances(t *testing.T) {
+	initiatorSigner := keypair.MustRandom()
+	responderSigner := keypair.MustRandom()
+
+	initiatorEscrow, err := keypair.ParseAddress("GDU5LGMB7QQPP5NABMPCI7JINHSEBJ576W7O5EFCTXUUZX63OJUFRNDI")
+	require.NoError(t, err)
+	responderEscrow, err := keypair.ParseAddress("GAWWANJAAOTAGEHCF7QD3Y5BAAIAWQ37323GKMI2ZKK34DJT2KX72MAF")
+	require.NoError(t, err)
+
+	initiatorEscrowAccount := &EscrowAccount{
+		Address:        initiatorEscrow.FromAddress(),
+		SequenceNumber: int64(101),
+	}
+	responderEscrowAccount := &EscrowAccount{
+		Address:        responderEscrow.FromAddress(),
+		SequenceNumber: int64(202),
+	}
+	initiatorChannel := NewChannel(Config{
+		NetworkPassphrase:   network.TestNetworkPassphrase,
+		MaxOpenExpiry:       time.Hour,
+		Initiator:           true,
+		LocalSigner:         initiatorSigner,
+		RemoteSigner:        responderSigner.FromAddress(),
+		LocalEscrowAccount:  initiatorEscrowAccount,
+		RemoteEscrowAccount: responderEscrowAccount,
+	})
+	initiatorChannel.UpdateLocalEscrowAccountBalance(10_000_0000000)
+	initiatorChannel.UpdateRemoteEscrowAccountBalance(10_000_0000000)
+
+	// Deposit, payment of 20 xlm to initiator escrow.
+	paymentResultMeta := "AAAAAgAAAAIAAAADABAqFAAAAAAAAAAAWPnYf+6kQN3t44vgesQdWh4JOOPj7aer852I7RJhtzAAAAAXHr20ywANrPwAAAAGAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAABABAqFAAAAAAAAAAAWPnYf+6kQN3t44vgesQdWh4JOOPj7aer852I7RJhtzAAAAAXHr20ywANrPwAAAAHAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAABAAAABAAAAAMAECn+AAAAAAAAAADp1ZmB/CD39aALHiR9KGnkQKe/9b7ukKKd6Uzf23JoWAAAABdIdugAABAp/gAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAEAECoUAAAAAAAAAADp1ZmB/CD39aALHiR9KGnkQKe/9b7ukKKd6Uzf23JoWAAAABdUYqoAABAp/gAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAMAECoUAAAAAAAAAABY+dh/7qRA3e3ji+B6xB1aHgk44+Ptp6vznYjtEmG3MAAAABcevbTLAA2s/AAAAAcAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAEAECoUAAAAAAAAAABY+dh/7qRA3e3ji+B6xB1aHgk44+Ptp6vznYjtEmG3MAAAABcS0fLLAA2s/AAAAAcAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAA="
```

Comment:
How did you generate these?

Reply:
from lab. And then if the balance was slightly off because of tx fees I converted to object, changed the balance, and then converted back to xdr to mimic as if it was fee bumped

Reply:
It might be worth adding to their comments the version of the protocol they were generated with, in this case v17. So in the future if we need to update them we know what might be different.

Reply:
good idea https://github.com/stellar/experimental-payment-channels/pull/234/commits/0be276227600a502ce1ccccfd420fe1962584ca1